### PR TITLE
feat(refs T35963): enable split statements 

### PR DESCRIPTION
**Pull request by gruenbergerdemos**
**Ticket:** https://yaits.demos-deutschland.de/T35963

We want to have the feature of splitting statements in the core, too, but without the pipeline. 

- Moved all necessary file from addon to core (all but the processing page)
- add `splitStatementView` directly in twig instead of `addonWrapper`
- `addonWrapper` (hook: split.statement.ai) is now in `SplitStatementView`
- `addonWrapper` now emits an event with boolean payload if it has loaded successfully or not
- `addonWrapper` forwards `segementationStatus:change` event from addon component
- changes in `SplitStatementView`:
1. Listen to loaded event from addon. If not loaded (no addon installed) `fetchSegmentDraftList` and `fetchInitialData`. If there is no segmentation yet and no segmentDraftList `setInitialData` for SegmentDraftList and `fetchTags`. 
2. Listen to `segmentationStatus:change` event forwared from addon and `setSegmentationStatus`
3. Remove `fetchSegmentDraftList` and `fetchInitialData` from mounted(), because we need to distinguish if addon is installed or not. Fetching segments with ai is handled in Addon. 
- In store add action for initializing data without segmentDraftList form backend and an own action for fetchTags. 

**Open questions:**

- We need to listen to event of the `SplitStatementAi`. Therefor the `AddonWrapper` needs to forward the event. I'm not sure about this solution, because the `AddonWrapper` is a generic component for all addons and it feels weird to forward an event from a specific component. Any ideas? My only other idea was to use vue bus.
- To check if an addon was loaded or not (aka is installed or not) I added an event `isLoaded` to the `AddonWrapper` which is always emitted and has a boolean as payload (true = addon has loaded, false = addon has not loaded). Is there any other way to check if an addon is installed or do we want to do it like this?
- I moved the logic for getting data from the pipeline to the addon in `SplitStatementAi` since we can access a core store from the addon. The component emits an event for changing the segmentationStatus. We can use this for the click event, too, so we don't need to forward two different events. The only pipeline logic in core is now in `SplitStatementStore` (like before) and some prosemirror stuff (in `/lib/prosemirror`). Do we leave it like this? I'm not sure how we can separate this. 

### How to review/test
- Install addon with correct branch
- You should be able to split statements with addon enabled or disabled
- Addon enabled: Should work life before
- Addon disabled: You should get to the split statement view immediately without the processing page. There everything should work as usual.

### CODE REVIEW

File: client/js/components/addon/AddonWrapper.vue [hunk 1]
```diff
@@ -75,7 +76,11 @@ export default {
             this.$options.components[addon.entry] = window[addon.entry].default
 
             this.component = window[addon.entry].default
+
+            this.isLoaded = true
           }
+
+          this.$emit('loaded', this.isLoaded)
```
Comment (spiess-demos) [inline client/js/components/addon/AddonWrapper.vue:83]: As i understand it, if `addons.assets.load` results in no response (aka there is no addon installed which uses the hook), the payload of the 'loaded' event will be `false`, which can be used by the parent component to apply its logic. this is a sufficient solution imho.

**Review by spiess-demos — APPROVED** [review]
Lgtm (code review).
Thanks for this PR, looks great!

As for the StatementMeta and related components, did you diff the files you deleted within the other PR with the already-existing files in demosplan-core, to prevent dropping fixes that already exist in the addon?

File: client/js/bundles/procedure/administrationSplitStatement.js [hunk 2]
```diff
@@ -11,12 +11,12 @@
  * This is the entrypoint for administration_split_statement.html.twig
  */
 
-import AddonWrapper from "@DpJs/components/addon/AddonWrapper"
 import { initialize } from '@DpJs/InitVue'
 import SplitStatementStore from '@DpJs/store/procedure/SplitStatementStore'
+import SplitStatementView from '@DpJs/components/statement/splitStatement/SplitStatementView'
```
[the remark this answers is not in the document]
Comment (gruenbergerdemos) [reply client/js/bundles/procedure/administrationSplitStatement.js:16]: Done in [6761516](https://github.com/demos-europe/demosplan-core/pull/2816/commits/676151687befb46909250beb35e7f4cfdbe91002)

File: client/js/components/addon/AddonWrapper.vue [hunk 3]
```diff
@@ -4,7 +4,7 @@
       :is="component"
       :ref="refComponent"
       v-bind="addonProps"
-    />
+      @segmentationStatus:change="(status) => this.$emit('segmentationStatus:change', status)" />
```
[the remark this answers is not in the document]
Comment (gruenbergerdemos) [reply client/js/components/addon/AddonWrapper.vue]: I talked with @salisdemos and we had the idea to make a more general event for the wrapper and give the specific event name as payload in the event. The addon wrapper then forwards the event with the name and rest of payload. What do you think about this approach? Done in [b0c4de4](https://github.com/demos-europe/demosplan-core/pull/2816/commits/b0c4de4518d4aa4995bbdbccae217fc67ed23a90)

File: client/js/components/addon/AddonWrapper.vue (hunk 1 above)
Comment (gruenbergerdemos) [reply client/js/components/addon/AddonWrapper.vue]: Yes, that was the idea. But @salisdemos mentioned that we can load more than one addon per hook, so I changed it. Now the event emits an array with all loaded addons and in the parent you can check if a specific addon was loaded (with the entry name). Done in [b0c4de4](https://github.com/demos-europe/demosplan-core/pull/2816/commits/b0c4de4518d4aa4995bbdbccae217fc67ed23a90)

**Comment by gruenbergerdemos** [comment]
> As for the StatementMeta and related components, did you diff the files you deleted within the other PR with the already-existing files in demosplan-core, to prevent dropping fixes that already exist in the addon?

No I didn't think of that, thanks for the reminder. There was only one small change done in the addon, see here [9d80825](https://github.com/demos-europe/demosplan-core/pull/2816/commits/9d8082557946ed8510f78cce6b459569ef90771d)

File: client/js/components/statement/splitStatement/CardPaneCard.vue [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      <span class="color--grey">
+        {{ assigneeName }}
+      </span>
+    </p>
+
+    <dp-button-icon
+      icon="fa-pencil"
+      :text="Translator.trans('segment.edit')"
+      @click="$emit('edit-segment', segment.id)" />
+    <dp-button-icon
```
[the remark this answers is not in the document]
Comment (gruenbergerdemos) [reply client/js/components/statement/splitStatement/CardPaneCard.vue]: Thanks for the input. I moved the button to the addon with its own hook, done here [8b313d2](https://github.com/demos-europe/demosplan-core/pull/2816/commits/8b313d23a8005552a85a69a2716dc2a49951feed)

File: client/js/components/addon/AddonWrapper.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (spiess-demos) [reply client/js/components/addon/AddonWrapper.vue]: it is definitely an improvement!

**Review by spiess-demos — DISMISSED** [review]
lgtm, thanks for the PR! (code review)

**Pull request merged**